### ISSUE TITLE
Update monitor task to work with Tekton 0.9

### ISSUE THREAD

**Issue by a-roberts** [tektoncd/experimental#419]
The PullRequest resource has changed since 0.8 so under this item check that the monitor task works (I'm 99% sure it won't, we rely on certain files being in certain places) and then do a PR - helpfully it's now just to the one place

Hint: https://github.com/tektoncd/pipeline/releases/download/v0.9.2/release.yaml, reopen an existing PR, watch the monitor task and see what happens

Update: can't install the 0.9.2 release yaml on OCP 4.2, mentions an invalid image URL (fine on Docker Desktop), so I'll do the testing on my laptop instead for now

**Comment by a-roberts** [comment]
This is the error

```
(base) Adams-MBP:~ aroberts$ klc monitor-taskrun-nwzgq-pod-23e64e 
{"level":"info","ts":1578389086.162878,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: open /var/run/ko/refs/heads/pr: no such file or directory"}
{"level":"info","ts":1578389086.1631095,"logger":"fallback-logger","caller":"pullrequest-init/main.go:54","msg":"RUNNING DOWNLOAD!","resource_type":"pullrequest","mode":"download"}
{"level":"info","ts":1578389086.1632907,"logger":"fallback-logger","caller":"pullrequest/api.go:55","msg":"finding pr","resource_type":"pullrequest","mode":"download","provider":"github","owner":"AROBERTS","repo":"devops-world-example","pr":"32"}
finding pr%!(EXTRA int=32): Not Found
```

**Comment by a-roberts** [comment]
Spoke with @wlynch about this and we've got https://github.com/tektoncd/pipeline/issues/1816 created 🎉 including for GitLab which @dibbles will find useful. 

I think we're stuck on this until that's delivered short of changing to use the REST API (which we already do in places), I'd rather hold off on doing this until we have something to try from the Tekton side though.

**Comment by a-roberts** [comment]
@wlynch's change is good I think, it's trying to use github.ibm.com correctly now 

I've had to change the triggertemplate (in code it would be in the monitor task though) to use `authToken` not `githubToken` and now I get

```
:"32"}
{"level":"info","ts":1578489745.065222,"logger":"fallback-logger","caller":"pullrequest/api.go:61","msg":"finding combined status","resource_type":"pullrequest","mode":"download","provider":"github","owner":"AROBERTS","repo":"devops-world-example","pr":"32"}
{"level":"info","ts":1578489745.2085242,"logger":"fallback-logger","caller":"pullrequest/api.go:72","msg":"finding comments: %v&{0xc000184400 AROBERTS/devops-world-example 32 0xc00010e5b8}","resource_type":"pullrequest","mode":"download","provider":"github","owner":"AROBERTS","repo":"devops-world-example","pr":"32"}
{"level":"info","ts":1578489745.4254239,"logger":"fallback-logger","caller":"pullrequest/api.go:77","msg":"found comments: %v[0xc0006240e0 0xc0006241c0 0xc0006242a0 0xc000624380 0xc000624460 0xc000624540 0xc000624620 0xc000624700 0xc0006247e0 0xc0006248c0 0xc0006249a0 0xc000624a80 0xc000624b60 0xc000624c40 0xc000624d20 0xc000624e00 0xc000624ee0 0xc000624fc0 0xc0006250a0 0xc000625180 0xc000625260 0xc000625340 0xc000625420 0xc000625500 0xc0006255e0 0xc0006256c0 0xc0006257a0 0xc000625880 0xc000625960 0xc000625a40]","resource_type":"pullrequest","mode":"download","provider":"github","owner":"AROBERTS","repo":"devops-world-example","pr":"32"}
Traceback (most recent call last):
  File "<stdin>", line 14, in <module>
IOError: [Errno 2] No such file or directory: '/workspace/pull-request/github/pr.json'
2020/01/08 13:22:27 Skipping step because a previous step failed
```

which I'll fix next.

**Comment by a-roberts** [comment]
Fix is at https://github.com/tektoncd/experimental/pull/425

This should only be merged when we no longer support Tekton 0.8 and requires master of Tekton Pipelines (so something later than 0.9.2).

@mnuttall FYI

**Comment by a-roberts** [comment]
Change were consumed by https://github.com/tektoncd/experimental/pull/434 so closing (this is done), we actually rely on 0.10 now
